Bio/SeqIO/AbiIO.py

```diff
@@ -360,15 +359,16 @@ def AbiIterator(handle, alphabet=None, trim=False):
         if isinstance(Alphabet._get_base_alphabet(alphabet), Alphabet.RNAAlphabet):
             raise ValueError("Invalid alphabet, ABI files do not hold RNA.")
 
-    # raise exception if handle mode is not 'rb'
-    if hasattr(handle, "mode"):
-        if set("rb") != set(handle.mode.lower()):
-            raise ValueError("ABI files has to be opened in 'rb' mode.")
+    try:
+        handle = open(argument, "rb")
+    except TypeError:
+        handle = argument
+        if handle.read(0) != b"":
+            raise ValueError("ABI files has to be opened in 'rb' mode.") from None
```

Comment:
Pre-existing typo, has/have.

Other file formats use something like "ABI files must be opened in binary mode." but we're not being entirely consistent here.